test/e2e/kubelet_stats.go

```diff
@@ -238,12 +247,14 @@ func getOneTimeResourceUsageOnNode(c *client.Client, nodeName string, cpuInterva
 		return nil, err
 	}
 	// Process container infos that are relevant to us.
-	containers := targetContainers()
-	usageMap := make(map[string]*containerResourceUsage, len(containers))
+	containers := containerNames()
+	usageMap := make(resourceUsagePerContainer, len(containers))
 	for _, name := range containers {
 		info, ok := containerInfos[name]
-		if !ok {
+		if !ok && !expectMissingContainers {
```

Comment:
I think that doing:

if !ok {
  if !expectMissingContainers {
    return nil, fmt.Errorf(...)
  }
  continue
}

is more readable.